codec-http/src/main/java/io/netty/handler/codec/spdy/SpdySession.java

```diff
@@ -322,7 +323,8 @@ void clearPendingWrites(Throwable cause) {
         }
     }
 
-    private final class StreamComparator implements Comparator<Integer> {
+    private final class StreamComparator implements Comparator<Integer>, Serializable {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Make this inner class static [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2066)

Reply:
Hm... If I will follow this rule, I have to make Serializable a huge chain of classes: StreamComparator -> SpdySession -> StreamState -> PendingWrite -> ChannelPromise...

We can remove back `Serializable` from `StreamComparator`, because it used only once - inside method `SpdySession.activeStreams()` and after that will never be serialized. So, in this case, it's not a critical problem.